docs/design/tag-matching.md

```diff
@@ -0,0 +1,72 @@
+
+# Support for "Tag Matching" when placing bricks
+
+
+## Use case
+
+Users frequently request creating different performance teirs or service
+levels using Heketi managed Gluster. Currently, this is not possible
+unless the devices are split across different nodes such that nodes with
+device types X form one Gluster TSP and nodes with device type Y form
+a different Gluster TSP. Then Heketi can be requested to allocate volumes
+from one cluster or another. If one wants to deploy gluster on bare-metal
+with a mix of device types this workaround fails.
+
+## Summary
+
+This design proposes a simple "tag matching" approach to require Heketi
+only use certain devices when placing a volume. To keep the design simple
+no sort of auto-detection will be performed. The user is required to
+specify a key-value pair on a specific set of devices. Then a new
+volume option with a "user.heketi.tag-match" key and a simple matching
+rule (see below) must be specified when the volume is created.
+
+Then Heketi will ensure that bricks for this volume will only be
+placed on devices where the tag matching rule applies.
+
+This approach largely re-uses existing mechanisms established during the
+development of the arbiter feature as well as the zone-checking
+features.
+
+## Workflow Example
+
+Assume three nodes each with three existing devices each. One of the
+devices on the node is a smaller high-performance device. The
+remaining two devices are large slower devices.
+
+For each device on one example node:
+```
+   $ heketi-cli device settags abc1230901344aef90 tier:gold
+   $ heketi-cli device settags bbc1230901344aef91 tier:silver
+   $ heketi-cli device settags cbc1230901344aef92 tier:silver
+```
+
+Now that devices have been tagged a volume can be requested to
+make use of the tagging:
+```
+    $ heketi-cli volume create --size=5 --gluster-volume-options 'user.heketi.tag-match tier=gold'
+    $ heketi-cli volume create --size=100 --gluster-volume-options 'user.heketi.tag-match tier=silver'
```

Comment:
I am thinking we need to have device in the key for example `user.heketi.device-tag-match` because we might later want to match only some nodes with more CPU etc(we already have for zone).

Reply:
OK, sounds good to me.